src/components/select/option/bl-select-option.ts

```diff
@@ -0,0 +1,76 @@
+import { LitElement, html, CSSResultGroup } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { event, EventDispatcher } from '../../../utilities/event';
+import { ISelectOption } from '../types';
+import style from './bl-select-option.css';
+
+@customElement('bl-select-option')
+export default class BlSelectOption extends LitElement {
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  /* Declare reactive properties */
+  /**
+   * Sets the value for the option
+   */
+  @property({})
+  value: string;
+
+  /**
+   * Sets option as disabled
+   */
+  @property({ type: Boolean })
+  disabled?: boolean = false;
+
+  /**
+   * Sets option as selected state
+   */
+  @property({ type: Boolean, reflect: true })
+  selected?: boolean = false;
+
+  @state()
+  isCheckbox?: boolean = false;
```

Comment:
There is a problem here but I need some more time to explain it 😊

I tried, I wrote many things but then removed all. First I need to evaluate some options on my mind. I'll come back here soon.